apps/andi/lib/andi/input_schemas/dataset_input.ex

```diff
@@ -0,0 +1,142 @@
+defmodule Andi.InputSchemas.DatasetInput do
+  @moduledoc """
+  Module for validating Ecto.Changesets on flattened dataset input.
+  """
+  import Ecto.Changeset
+
+  alias Andi.Services.DatasetRetrieval
+  alias Andi.InputSchemas.DatasetSchemaValidator
+  alias Andi.InputSchemas.DisplayNames
+
+  @business_fields %{
+    contactEmail: :string,
+    contactName: :string,
+    dataTitle: :string,
+    description: :string,
+    homepage: :string,
+    issuedDate: :date,
+    keywords: {:array, :string},
+    language: :string,
+    license: :string,
+    modifiedDate: :date,
+    orgTitle: :string,
+    publishFrequency: :string,
+    spatial: :string,
+    temporal: :string
+  }
+
+  @technical_fields %{
+    dataName: :string,
+    orgName: :string,
+    private: :boolean,
+    schema: {:array, :map},
+    sourceFormat: :string,
+    sourceType: :string,
+    topLevelSelector: :string
+  }
+
+  @types %{id: :string}
+         |> Map.merge(@business_fields)
+         |> Map.merge(@technical_fields)
+
+  @required_fields [
+    :contactEmail,
+    :contactName,
+    :dataName,
+    :dataTitle,
+    :description,
+    :issuedDate,
+    :license,
+    :orgName,
+    :orgTitle,
+    :private,
+    :publishFrequency,
+    :sourceFormat,
+    :sourceType
+  ]
+
+  @email_regex ~r/^[A-Za-z0-9._%+-+']+@[A-Za-z0-9.-]+\.[A-Za-z]+$/
+  @no_dashes_regex ~r/^[^\-]+$/
+
+  def business_keys(), do: Map.keys(@business_fields)
+  def technical_keys(), do: Map.keys(@technical_fields)
+  def all_keys(), do: Map.keys(@types)
+
+  def get_display_name(field_key), do: DisplayNames.get(field_key)
```

Comment:
This seems like it will be confusing for errors that come through on the API. I'm wondering if there's a way to do this message conversion on the view (Maybe in the error_view?).

Reply:
I guess I'd only argue that it won't be confusing because the format of the response will still look like this:
```
{ "errors": { "contactEmail": ["Please provide a valid maintainer email."] } }
```

Reply:
Checkout AndiWeb.ErrorHelpers

Reply:
Trying to think about how to build these messages in the view... we'd need to create some kind of mapping from error messages from the changeset validation to error message on the UI.  Seems like a layer of abstraction I'd prefer to avoid for now.